src/kakarot/instructions/system_operations.cairo

```diff
@@ -301,6 +301,60 @@ namespace SystemOperations {
 
         return sub_ctx;
     }
+
+    // @notice SELFDESTRUCT operation.
+    // @dev
+    // @custom:since Frontier
+    // @custom:group System Operations
+    // @custom:gas 3000 + dynamic gas
+    // @custom:stack_consumed_elements 1
+    // @return The pointer to the updated execution_context.
+    func exec_selfdestruct{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr: BitwiseBuiltin*,
+    }(ctx: model.ExecutionContext*) -> model.ExecutionContext* {
+        alloc_locals;
+
+        // Get stack and memory from context
+        let stack = ctx.stack;
+
+        // Stack input:
+        // 0 - address: account to send the current balance to
+        let (stack, address_uint256) = Stack.pop(stack);
+        let address_felt = Helpers.uint256_to_felt(address_uint256);
+        let ctx = ExecutionContext.update_stack(ctx, stack);
+        // Get the number of native tokens owned by the given starknet
+        // account and transfer them to receiver
+        let (native_token_address_) = native_token_address.read();
+        let (balance: Uint256) = IEth.balanceOf(
+            contract_address=native_token_address_, account=ctx.starknet_contract_address
+        );
+        let (success) = IEth.transfer(contract_address=native_token_address_, recipient=address_felt, amount=balance);
+        with_attr error_message("Kakarot: Transfer failed") {
+            assert success = TRUE;
+        }
+
+        // Save contract to be destroyed at the end of the transaction in a NEW array
+        // Directly filling ctx.destroy_contract produces error "Expected memory address to be relocatable value. Found: 0."
+        let (destroy_contracts) = alloc();
+        Helpers.fill_array(
+            fill_len=ctx.destroy_contracts_len,
+            input_arr=ctx.destroy_contracts,
+            output_arr=destroy_contracts,
+        );
+        assert [destroy_contracts + ctx.destroy_contracts_len] = ctx.starknet_contract_address;
+        
+        let ctx = ExecutionContext.update_destroy_contracts(
+            self=ctx,
+            destroy_contracts_len= ctx.destroy_contracts_len + 1,
+            destroy_contracts=destroy_contracts,
+            stop=TRUE,
```

Comment:
Do we really stop the execution context in a SELFDESTRUCT ?

Reply:
I think so as the opcode description is "**Halt execution** and register account for later deletion", which is similar to STOP opcode "**Halts execution**"